flutter/lib/backend/bridge/ffi_run.dart

```diff
@@ -75,25 +76,13 @@ class _RunOut extends Struct {
   @Int32()
   external int ok;
   @Float()
-  external double latency;
+  external double score;
```

Comment:
The name `score` is too general in this case. `latency` was more accurate, since what we calculate here is the number of query per second (so actually `1 / latency`). To be precise, we can name this `speed`, since that is what the benchmark does, measure `accuracy` and `speed`.

Reply:
Originally we had field `latency`, but we actually transferred both latency and score, depending on the benchmark, and then we transformed it differently, depending on type of benchmark.
This is caused by loadgen. Loadgen is used by C++ code, and Flutter doesn't even know about it, so it looked weird to me  that we had logic handling weird loadgen stuff in Flutter, and I decided to move this logic to C++ code.

>To be precise, we can name this speed, since that is what the benchmark does, measure accuracy and speed.

We literary have code like `objectA.score = objectB.score` in several places in code, without modifying the value, without any logic around it.
Where would we switch from using `speed` to using `score`?
We should either use `score` everywhere, or use `speed` everywhere, or else the place where we change the name will be very strange.

`score` is used everywhere in context of different benchmarks, so I personally think we should stick with it.

Reply:
> Originally we had field latency, but we actually transferred both latency and score, depending on the benchmark, and then we transformed it differently, depending on type of benchmark.

Can you give an example? I see only latency transferred in `cpp/flutter/main.cc`
```C++
  if (std::string{in->scenario} == "Offline") {
    out->score = driver.ComputeLatency();
  } else {
    out->score = 1000.0 / driver.ComputeLatency();
  }
```

Reply:
Here you go:
https://github.com/mlcommons/mobile_app_open/blob/8b16c1baa07598160d37733cd0190732bb621855/flutter/lib/backend/bridge/ffi_run.dart#L115
https://github.com/mlcommons/mobile_app_open/blob/8b16c1baa07598160d37733cd0190732bb621855/flutter/lib/benchmark/state.dart#L253
https://github.com/mlcommons/mobile_app_open/blob/8b16c1baa07598160d37733cd0190732bb621855/flutter/lib/resources/result_manager.dart#L111-L113
https://github.com/mlcommons/mobile_app_open/blob/8b16c1baa07598160d37733cd0190732bb621855/flutter/lib/ui/result_screen.dart#L75-L77
We also do this when [serializing BenchmarkResult class](https://github.com/mlcommons/mobile_app_open/blob/8b16c1baa07598160d37733cd0190732bb621855/flutter/lib/benchmark/benchmark.dart#L78), but this is irrelevant because we can change internal format for results at any time.
However, when writing `results.json` we certainly write `score` field, and I'm pretty sure everyone calls numbers on the result screen "scores". And we have `dart_ffi_run_benchmark_out.score == RunResult.score == BenchmarkResult.score`, and `RunResult.score == results.json[score]`, and `BenchmarkResult.score → number showed on result screen`.

I just don't see where we could change field name in this chain without making it confusing.

Reply:
Previously this logic was in the Flutter code.
This is what we currently have in the master branch:
https://github.com/mlcommons/mobile_app_open/blob/857fe9cdd6e49528ab3df53861219a7a0ff677e6/flutter/lib/benchmark/benchmark_result.dart#L51
And yeah, this is not much code, but it is still specific to loadgen.
We could technically keep the `latency` field, but I don't see a point if the result would be:
```c++
out->latency = 1000 / driver.ComputeLatency(); // if offline mode
```
```flutter
score = 1000 / out.ref.latency;
```

Reply:
Though, I'm fine with keeping `latency` field, it would still be easy to read.
I just don't see why it would be better than just transferring `score` field.

Reply:
* In performance mode, the score displayed is actually `1 / latency` or query per second (QPS). This should be the same for any benchmarks.
* In accuracy mode, the score displayed is the accuracy of each benchmark. This will be different for each benchmark, since each benchmark has a unique method to calculate accuracy.
So I think `speed` would be more accurate than `score`, or we can keep the `latency` name.

I guest the confusion with the 1000 is a legacy thing. It looks like the app sometimes showed `latency in ms` or `qps` (= `1000 / latency in ms`). But now they all show in `qps`, that is why the conversion are all over the place.
https://github.com/mlcommons/mobile_app_open/blob/857fe9cdd6e49528ab3df53861219a7a0ff677e6/android/cpp/mlperf_driver.h#L78-L90

If we want to fix it, then we should change it here and make this function return only `qps`
https://github.com/mlcommons/mobile_app_open/blob/857fe9cdd6e49528ab3df53861219a7a0ff677e6/android/cpp/mlperf_driver.h#L66-L75

Reply:
I see.

Changes in mlperf driver would affect the old Android app.
I do remember that we have an [issue](https://github.com/mlcommons/mobile_app_open/issues/153) to deprecate the old app, including separating its C++ code from Flutter, but Relja has explicitly asked us to delay it.

I suggest right now we just rename field `score` to `qps`, and then maybe do C++ code refactoring after old app deprecation.

Reply:
`qps` is a unit of measurement similar like `ms`. So I prefer to name the variable `speed` or `latency`.

Reply:
Ok, I changed qps to latency

Reply:
I see that `score` is still used instead of `latency` in many places, e.g. in `BenchmarkResult`, `RunResult`.

Reply:
Because it it was always this way. I only renamed that field in C++ struct, so I reverted this change. In all other places I just moved what we had to different files, but I kept the original naming.
We also use `score` in tasks file:
https://github.com/mlcommons/mobile_models/blob/dec215a473e78e52d0d21806a8f24875bb50aca4/v1_0/assets/tasks_v3.pbtxt#L26
We also use `score`+`accuracy` in `result.json`:
https://github.com/mlcommons/mobile_app_open/blob/455e69fcbea2474ac238b2223ee32f7e652b617c/flutter/lib/resources/result_manager.dart#L79-L80
I believe that field names in `result.json` must never change (or at least an approval is required).

Reply:
I think we should discuss changing the name in a separate issue.

Reply:
Ok.